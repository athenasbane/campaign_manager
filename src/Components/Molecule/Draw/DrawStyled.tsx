import SwipeableDrawer from "@mui/material/SwipeableDrawer";
import Box from "@mui/material/Box";

import styled from "@mui/material/styles/styled";

export const StyledSwipeableDrawer = styled(SwipeableDrawer)(
  ({ theme: _ }) => ({
    zIndex: 1401,
  })
);

export const StyledBox = styled(Box)(({ theme: _ }) => ({
  width: "auto",
}));
