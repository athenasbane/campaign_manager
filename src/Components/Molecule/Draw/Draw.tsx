import List from "@mui/material/List";
import Divider from "@mui/material/Divider";
import MenuBookIcon from "@mui/icons-material/MenuBook";
import DownloadIcon from "@mui/icons-material/Download";
import AssignmentIcon from "@mui/icons-material/Assignment";
import MapIcon from "@mui/icons-material/Map";
import HistoryEduIcon from "@mui/icons-material/HistoryEdu";
import VisibilityIcon from "@mui/icons-material/Visibility";
import { EnumModalSlice } from "../../../Store/slices/modals";
import { useNavigate } from "react-router-dom";
import { BookOutlined } from "@mui/icons-material";
import { IDrawListItem } from "../../../Types/Interfaces/DrawListItem";
import DrawItem from "../DrawItem/DrawItem";
import { StyledBox, StyledSwipeableDrawer } from "./DrawStyled";

export interface IDrawProps {
  open: boolean;
  closeModal: () => void;
  openSingleModal: (modal: EnumModalSlice) => void;
}

export default function Draw({
  open,
  openSingleModal,
  closeModal,
}: IDrawProps) {
  const navigate = useNavigate();

  /**
   * [TODO]
   * Move these to contentful
   * Will need to create a icon component that has a
   * Dropdown in contentful that uses the selected Icon
   */
  const topListItems: IDrawListItem[] = [
    {
      displayText: "Maps",
      path: "/list/4bOnoLNoNujSq6sWnG6BEt",
      icon: <MapIcon />,
    },
    {
      displayText: "Session Recap",
      path: "/sessions",
      icon: <VisibilityIcon />,
    },
    {
      displayText: "Missions",
      path: "/missions",
      icon: <AssignmentIcon />,
    },
  ];

  const bottomListItems: IDrawListItem[] = [
    {
      displayText: "Lore",
      path: "/list/6FoUmw8ML88eBJ1fSHO0A8",
      icon: <MenuBookIcon />,
    },
    {
      displayText: "Tales from Teratin",
      path: "/list/1iLinkQTnQ1Q9qqzLGW37v",
      icon: <BookOutlined />,
    },
    {
      displayText: "History",
      path: "/history",
      icon: <HistoryEduIcon />,
    },

    {
      displayText: "Homebrew Documents",
      path: "/documents",
      icon: <DownloadIcon />,
    },
  ];

  return (
    <div>
      <StyledSwipeableDrawer
        anchor={"bottom"}
        open={open}
        onClose={closeModal}
        onOpen={() => openSingleModal(EnumModalSlice.Menu)}
      >
        <StyledBox
          role="presentation"
          onClick={closeModal}
          onKeyDown={closeModal}
        >
          <List>
            {topListItems.map((item) => (
              <DrawItem
                key={item.displayText}
                item={item}
                onClick={() => navigate(item.path)}
              />
            ))}
          </List>
          <Divider />
          <List>
            {bottomListItems.map((item) => (
              <DrawItem
                key={item.displayText}
                item={item}
                onClick={() => navigate(item.path)}
              />
            ))}
          </List>
        </StyledBox>
      </StyledSwipeableDrawer>
    </div>
  );
}
