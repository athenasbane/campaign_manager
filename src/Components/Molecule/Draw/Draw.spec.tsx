import { render } from "@testing-library/react";
import Draw from "Components/Molecule/Draw/Draw";
import { BrowserRouter } from "react-router-dom";

describe("Molecule - Draw", () => {
  const closeModalStub = jest.fn();
  const openSingleModalStub = jest.fn();
  it("should match snapshot", () => {
    const { container } = render(
      <Draw
        open={true}
        closeModal={closeModalStub}
        openSingleModal={openSingleModalStub}
      />,
      { wrapper: BrowserRouter }
    );

    expect(container).toMatchSnapshot();
  });
});
