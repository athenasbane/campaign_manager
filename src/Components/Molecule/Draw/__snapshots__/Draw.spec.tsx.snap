// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecule - Draw should match snapshot 1`] = `
<div
  aria-hidden="true"
>
  <div>
    <div
      class="PrivateSwipeArea-root css-17j2vre"
      style="height: 20px;"
    />
  </div>
</div>
`;
